native/cocos/platform/java/modules/SystemWindowManager.cpp

```diff
@@ -0,0 +1,53 @@
+/****************************************************************************
+ Copyright (c) 2021-2022 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#include "SystemWindowManager.h"
+#include "BasePlatform.h"
+#include "platform/java/modules/SystemWindow.h"
+
+namespace cc {
+
+SystemWindowManager *SystemWindowManager::_instance = nullptr;
+
+ SystemWindowManager::SystemWindowManager(IEventDispatch *delegate) {
+    _instance = this;
+}
+
+int SystemWindowManager::init() {
+    return 0;
+}
+
+void SystemWindowManager::poolEvent(bool *quit) {
+}
+
+void SystemWindowManager::swapWindows() {
+}
+
+ISystemWindow *SystemWindowManager::createWindow(const char *name) {
+    auto window = BasePlatform::getPlatform()->createNativeWindow();
+    _windows.insert(std::make_pair(std::string(name), window));

```

Comment:
Could simplify code with `_windows[name] = window;`?

Reply:
It's fine too，`insert ` can preventing overwrite an existing value with a same key?

Reply:
Yes, insert will not succeed if the key has already existed.
https://en.cppreference.com/w/cpp/container/unordered_map/insert

> Inserts element(s) into the container, if the container doesn't already contain an element with an equivalent key.